Fix #673 : Add utility for BCP47 normalisation

ansell opened the pull request:
This PR addresses GitHub issue: #673 .

Briefly describe the changes proposed in this PR:

* Adds a utility function for the steps required for BCP47 language tag normalisation
* Converts BCP47LanguageHandler to use the utility function

Signed-off-by: Peter Ansell <p_ansell@yahoo.com>

jeenbroekstra commented on core/model/src/main/java/org/eclipse/rdf4j/model/util/Literals.java:
```diff
@@ -580,6 +582,22 @@ public static boolean isLanguageLiteral(Literal literal) {
+	 * @throws IllformedLocaleException
+	 *         If the given language tag is illformed according to the rules specified in BCP47.
+	 */
+	public static String normaliseBCP47Tag(String languageTag)
```
Given that BCP47 is the "format to follow" according to the RDF 1.1 spec, could we simplify the method name to just `normaliseTag`?

ansell replied:
I would prefer not to remove BCP47 without adding Language in its place, if that makes sense. 

On another note, I have just noticed that I used the Australian English spelling of normalise, instead of the US English that we use in the rest of the API, normalize, so I will need to change that to stay consistent.

normalizeLanguageTag?

jeenbroekstra replied:
Just realized that giving a 👍 probably doesn't notify you, so: your proposal sounds good to me.

ansell commented:
I changed the name to normalizeLanguageTag, and also added deprecation tags for some ancient RFC3066 code that we have sitting around that wasn't being maintained. We will now rely on the JDK to maintain the language tag registry, through the Locale APIs.

jeenbroekstra commented:
Hm, looking at that older code: I am ashamed to admit that I had no idea we even have a `LanguageTag` class (or indeed a whole package with related utilities). It might be worth looking into that a bit further and see if it's useful to update it to work with BCP47.
